### ISSUE TITLE
Overlay or Cross "No data received"

### ISSUE THREAD

**Issue by superteece** [RUB-NDS/PRET#5]
Is the message "No data received" indicating that the attempt has failed? Nothing is printed on the subsequent jobs except for what is on the printed document.

**Comment by jensvoid** [comment]
No, the message just means that no data was returned by the printer, which is absolutely correct behavior in case of the "overlay" command. (However, We should definitely use more meaningful messages in future releases).

There are two potentials reasons, the attack did not work:

1. The printer is not vulnerable because the exitserver/startjob operators as described in the PostScript standard are not supported by the device to access subsequent jobs (e.g. Brother printers and some older Xerox printers). You can easily test this by permanently defining "foo":

    printer:/> site true 0 startjob /foo true def
    No data received.
    /dev/usb/lp0:/> site userdict /foo known ==
    true

If the printer responds with "false" instead, attacks like "overlay" or "capture" will not work.

2. Subsequent jobs have not been printed using PostScript as a printer driver. This is important, because PostScript-based attacks like "overlay" can only affect PostScript-based documents (and not PCL, for example). You can easily test this:

    printer:/> overlay overlays/smiley.eps
    printer:/> print testpages/lorem.ps

If you can see the smiley on this test printout, but not on your own print jobs, you are not using PostScript as a printer driver. The attack however will/can only work on PostScript-based documents.
